x/gov/migrations/v3/convert.go

```diff
@@ -54,9 +54,13 @@ func ConvertToLegacyProposal(proposal v1.Proposal) (v1beta1.Proposal, error) {
 				return v1beta1.Proposal{}, err
 			}
 			legacyProposal.Content = legacyMsg.Content
+			return legacyProposal, nil
 		}
 	}
-	return legacyProposal, nil
+	// hack to fill up the content with the first message
+	legacyProposal.Content, err = codectypes.NewAnyWithValue(msgs[0])
```

Comment:
why are we doing the first message? What if there are multiple?

Reply:
The current logic doesn't support that. See few lines above how the `Content` is assigned if there is a `MsgExecLegacyContent`: 
* previously, if there are few, then only the last one will be set (we overwrite)
* I added early return - so the first one will be set instead.

Reply:
I don't get it. If there are multiple messages, then you only use the first (assuming there exists no `MsgExecLegacyContent`).

Reply:
Correct. I'm not sure which part is not clear for you. I've added a second fallback to set the content. We can only set one content, and tools expect something like a "legacy" `b1beta1.Proposal.Content` (which is an interface, but they try to get title and description from it).